src/plugins/intel_gna/orientation_helper.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "orientation_helper.hpp"
+
+#include <legacy/ie_layers.h>
+
+#include <gna_graph_tools.hpp>
+
+namespace GNAPluginNS {
+namespace helpers {
+
+void updateModelInputOrientationWithoutConvolution(const InferenceEngine::CNNLayer& inputLayer,
+                                                   const GNAPluginNS::backend::DnnComponents& components,
+                                                   GNAPluginNS::GnaInputs& inputs) {
+    // does not make sense to go further is there is no input to set
+    auto input = inputs.find(inputLayer.name);
+
+    if (input == inputs.end()) {
+        return;
+    }
+
+    auto doesntHaveGnaMapping = [=](InferenceEngine::CNNLayerPtr l) {
+        auto dnnLayer = components.findComponent(l);
+        return dnnLayer == nullptr;
+    };
+
+    auto nextLayers = InferenceEngine::CNNNetGetAllNextLayersSkipCertain(&inputLayer, -1, doesntHaveGnaMapping);
+    if (nextLayers.empty()) {
+        input->orientation = kDnnInterleavedOrientation;
+        return;
+    }
+
+    std::vector<intel_dnn_orientation_t> suggestedOrientations;
+
+    auto dims = input->dims;
+    auto raws = dims[0];
+    auto rest = InferenceEngine::details::product(std::next(std::begin(dims)), std::end(dims));
+    if (rest == 0) {
+        rest = 1;
+    }
+
+    for (auto& nextLayer : nextLayers) {
+        auto dnnLayer = components.findComponent(nextLayer);
+
+        if (!dnnLayer) {
```

Comment:
BTW how it can happen if we filtered them already?

Reply:
theoretically It shouldn't, buy in this context I could not know that, so I want to avoid potential segfault in case method would be called in different context.

Reply:
can we reevaluate this comment?